src/fittable_laws.jl

```diff
@@ -77,3 +77,172 @@ function (law::FM90)(wave::T) where T
+        FUV_amp=18.54,
+        FUV_lambda=0.07,
+        FUV_b=4.0,
+        FUV_n=6.5,
+        NUV_amp=0.0596,
+        NUV_lambda=0.22,
+        NUV_b=-1.95,
+        NUV_n=2.0,
+        SIL1_amp=0.0026,
+        SIL1_lambda=9.7,
+        SIL1_b=-1.95,
+        SIL1_n=2.0,
+        SIL2_amp=0.0026,
+        SIL2_lambda=18.0,
+        SIL2_b=-1.8,
+        SIL2_n=2.0,
+        FIR_amp=0.0159,
+        FIR_lambda=25.0,
+        FIR_b=0.0,
+        FIR_n=2.0)
+
+Pei (1992) generative extinction model applicable from the extreme UV to far-IR.
+
+## Parameters
+
+Background Terms
+* `BKG_amp` - amplitude
+* `BKG_lambda` - central wavelength
+* `BKG_b` - b coefficient
+* `BKG_n` - n coefficient
+
+Far-Ultraviolet Terms
+* `FUV_amp` - amplitude
+* `FUV_lambda` - central wavelength
+* `FUV_b` - b coefficent
+* `FUV_n` - n coefficient
+
+Near-Ultraviolet (2175 Å) Terms
+* `NUV_amp` - amplitude
+* `NUV_lambda` - central wavelength
+* `NUV_b` - b coefficent
+* `NUV_n` - n coefficient
+
+1st Silicate Feature (~10 micron) Terms
+* `SIL1_amp` - amplitude
+* `SIL1_lambda` - central wavelength
+* `SIL1_b` - b coefficent
+* `SIL1_n` - n coefficient
+
+2nd Silicate Feature (~18 micron) Terms
+* `SIL2_amp` - amplitude
+* `SIL2_lambda` - central wavelength
+* `SIL2_b` - b coefficient
+* `SIL2_n` - n coefficient
+
+Far-Infrared Terms
+* `FIR_amp` - amplitude
+* `FIR_lambda` - central wavelength
+* `FIR_b` - b coefficent
+* `FIR_n` - n coefficient
+
+If `λ` is a `Unitful.Quantity` it will be automatically converted to Å and the returned value will be `UnitfulAstro.mag`.
+
+## Examples
+```jldoctest
+julia> model = P92();
+
+julia> model(1500)
+2.396291891812002
+
+julia> P92(FUV_b = 2.0).([1000, 2000, 3000])
+3-element Array{Float64,1}:
+ 3.8390886792306187
+ 2.7304534614548697
+ 1.806181164464396
+
+```
+
+## Default Parameter Values
+
+|Term |lambda|A|b|n|
+|:---:|:---:|:---:|:---:|:---:|
+|BKG  |0.047 |218.57142857142858   |90   |2  |
+|FUV  |0.07  |18.545454545454547   |4.0  |6.5|
+|NUV  |0.22  |0.05961038961038961  |-1.95|2.0|
+|SIL1 |9.7   |0.0026493506493506496|-1.95|2.0|
+|SIL2 |18.0  |0.0026493506493506496|-1.8 |2.0|
+|FIR  |25.0  |0.015896103896103898 |0.0  |2.0|
+
+
+## References
+[Pei (1992)](https://ui.adsabs.harvard.edu/abs/1992ApJ...395..130P)
+"""
+@with_kw struct P92{T<:Number} <: ExtinctionLaw @deftype T
+    BKG_amp = 165.0 * (1 / 3.08 + 1)
+    BKG_lambda = 0.047
+    BKG_b = 90.0
+    BKG_n = 2.0
+    FUV_amp = 14.0 * (1 / 3.08 + 1)
+    FUV_lambda = 0.07
+    FUV_b = 4.0
+    FUV_n = 6.5
+    NUV_amp = 0.045 * (1 / 3.08 + 1)
+    NUV_lambda = 0.22
+    NUV_b = -1.95
+    NUV_n = 2.0
+    SIL1_amp = 0.002 * (1 / 3.08 + 1)
+    SIL1_lambda = 9.7
+    SIL1_b = -1.95
+    SIL1_n = 2.0
+    SIL2_amp = 0.002 * (1 / 3.08 + 1)
+    SIL2_lambda = 18.0
+    SIL2_b = -1.80
+    SIL2_n = 2.0
+    FIR_amp = 0.012 * (1 / 3.08 + 1)
+    FIR_lambda = 25.0
+    FIR_b = 0.00
+    FIR_n = 2.0
+    @assert BKG_amp ≥ 0 "`BKG_amp` must be ≥ 0, got $BKG_amp"
```

Comment:
I'm almost wondering if it would be easier to represent this struct hierarchically with tuples or named tuples
```julia
struct Example
	BKG = (165.0 * (1/3.08 + 1), 0.047, etc...)
    # or
    FIR = (A=0.012 * (1 / 3.08 + 1), lambda=25.0, b=0.0
end
```

It's clear this struct has this hierarchical order and it seems organized a little better. The question is whether this is easier for the user or if it makes sense for the user. Perhaps we don't use this for the struct itself, but we add a constructor that allows this behavior, something like
```julia
P92(BKG, FUV, NUV, SIL1, SIL2, FIR) = P92(BKG..., FUB..., NUV..., SIL1..., SILL2..., FIR...)
```
We could make this a kwarg-only version, but we'd have to remove the `@with_kw` from the struct (which is fine, we'd just have to add the assertion checking to another constructor, about the same complexity/lines-of-code)

Also, AFAIK this should have no performance impact (tuples and namedtuples are static and have lots of compiler opts.

What do you think?

Reply:
Your point is correct, that would make the struct quite organized, but let's think from the user's perspective who is trying to switch from DustExtinction (python) to this place, things like this could be intimidating to the user. What do you think?

Reply:
We can have both. The user isn't necessarily going to interact with the struct directlyy, just with the _constructors_ and the _methods_. We can organize the struct however we want and organize the constructors however we want. It may be easier to not worry about it